engine/table/src/main/java/io/deephaven/engine/table/impl/locations/impl/TableLocationKeySafetyImpl.java

```diff
@@ -0,0 +1,65 @@
+package io.deephaven.engine.table.impl.locations.impl;
+
+import io.deephaven.engine.table.impl.locations.TableDataException;
+import io.deephaven.engine.table.impl.locations.TableLocationKey;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+
+final class TableLocationKeySafetyImpl<TLK extends TableLocationKey> implements TableLocationKeyFinder<TLK> {
```

Comment:
I'm a little bit skeptical of this whole endeavor. Also, it needs JavaDoc. I think we might be better off checking the partition safety downstream, say, in the `PollingTableLocationProvider`.

Reply:
I have verified all the exceptional cases get caught, so it does _something_.

Reply:
This is an package private class; that said, there is javadoc on `io.deephaven.engine.table.impl.locations.impl.TableLocationKeyFinder#safetyCheck`.

Reply:
None that specifies what the safety checks are/do.